readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -831,3 +835,108 @@ def test_command_oom_kill(self):
         self.assertEqual(
             str(cmd.output),
             u'Command killed due to excessive memory consumption\n')
+
+
+
+
+class TestAutoWipeEnvironment(TestCase):
+    fixtures = ['test_data']
+
+    def setUp(self):
+        self.pip = Project.objects.get(slug='pip')
+        self.version = self.pip.versions.get(slug='0.8')
+
+    def test_is_obsolete_without_env_json_file(self):
+        yaml_config = create_load()()[0]
+        config = ConfigWrapper(version=self.version, yaml_config=yaml_config)
+
+        with patch('os.path.exists') as exists:
+            exists.return_value = False
+            python_env = Virtualenv(
+                version=self.version,
+                build_env=None,
+                config=config,
+            )
+
+        self.assertFalse(python_env.is_obsolete)
+
+    def test_is_obsolete_with_invalid_env_json_file(self):
+        yaml_config = create_load()()[0]
+        config = ConfigWrapper(version=self.version, yaml_config=yaml_config)
+
+        with patch('os.path.exists') as exists:
+            exists.return_value = True
+            python_env = Virtualenv(
+                version=self.version,
+                build_env=None,
+                config=config,
+            )
+
+        self.assertFalse(python_env.is_obsolete)
+
+    def test_is_obsolete_with_json_different_python_version(self):
+        config_data = {
+            'build': {
+                'image': '2.0',
+            },
+            'python': {
+                'version': 2.7,
+            },
+        }
+        yaml_config = create_load(config_data)()[0]
+        config = ConfigWrapper(version=self.version, yaml_config=yaml_config)
+
+        python_env = Virtualenv(
+            version=self.version,
+            build_env=None,
+            config=config,
+        )
+        env_json_data = '{"build": {"image": "readthedocs/build:2.0"}, "python": {"version": 3.5}}'
+        with patch('os.path.exists') as exists, patch('readthedocs.doc_builder.python_environments.open', mock_open(read_data=env_json_data)) as _open:  # noqa
+            exists.return_value = True
+            self.assertTrue(python_env.is_obsolete)
+
+    @pytest.mark.xfail(reason='build.image is not being considered yet')
```

Comment:
to make this test pass, we need the PR that get the image from yaml merged... then we can remove this line and it should pass :)